### DISCUSSION TITLE
Separating the frontend part from the backend part?

### DISCUSSION THREAD

**Discussion by farooqkz** [pcworms/pcworms-template#5] [Ideas]
From what I understand now, if one wants to make changes to this template, they have to learn PHP and WP specific functions/classes, too. If we could separate the frontend part from the backend one, one who wants to contribute to frontend won't have to know PHP and WP. Not sure if it's possible or that it's a good idea.

Perhaps one approach is a SPA with WP's RESTful API.

**Comment by b-simjoo** [comment]
I don't think that it is a good idea.
1. In my opinion, this is a big and fundamental change and it is better not to be related to the current template, it can be a fork of this, but it will be completely different from the current one.
2. In my opinion, it makes the template heavy and slow on the user's side. Our server has more resources than a mobile phone and WordPress is well-optimized for such purposes.
3. I think learning PHP will be fun for you!
4. I think we will have big problems with plugins.

**Reply by farooqkz** [reply]
1. Maybe
2. Not necessary. If we do it with a framework optimized for mobile, we won't have a problem. Also the front-end could be cached and only data could be downloaded from the server.
3. No I don't think so.
4. I think we will...

**Comment by farooqkz** [comment]
Another approach is a separate PWA which uses the WP RESTful API and keeping the current one as well.

**Reply by b-simjoo** [reply]
The WP RESTful API was disabled by the plugin due to security issues.

**Reply by farooqkz** [reply]
In that case, I would like to disable that plugin due to metasecurity reasons.
